--- a/build_tools/build_ext.py
+++ b/build_tools/build_ext.py
@@ -62,7 +62,7 @@
             cmake_path,
             "-B",
             build_dir,
-            f"-DPython_EXECUTABLE={sys.executable}",
+            "-DPython_EXECUTABLE=$PYTHON",
             f"-DPython_INCLUDE_DIR={sysconfig.get_path('include')}",
             f"-DPython_SITEARCH={sysconfig.get_path('platlib')}",
             f"-DCMAKE_BUILD_TYPE={build_type}",
--- a/transformer_engine/common/CMakeLists.txt
+++ b/transformer_engine/common/CMakeLists.txt
@@ -89,8 +89,12 @@
 include(${CMAKE_CURRENT_SOURCE_DIR}/../../3rdparty/cudnn-frontend/cmake/cuDNN.cmake)
 #
-set(CUTLASS_INCLUDE_DIR
-  "${CMAKE_CURRENT_SOURCE_DIR}/../../3rdparty/cutlass/include")
-set(CUTLASS_TOOLS_INCLUDE_DIR
-  "${CMAKE_CURRENT_SOURCE_DIR}/../../3rdparty/cutlass/tools/util/include")
+if(NOT DEFINED CUTLASS_INCLUDE_DIR)
+  set(CUTLASS_INCLUDE_DIR
+    "${CMAKE_CURRENT_SOURCE_DIR}/../../3rdparty/cutlass/include")
+endif()
+if(NOT DEFINED CUTLASS_TOOLS_INCLUDE_DIR)
+  set(CUTLASS_TOOLS_INCLUDE_DIR
+    "${CMAKE_CURRENT_SOURCE_DIR}/../../3rdparty/cutlass/tools/util/include")
+endif()
 #
 # Python
