--- a/transformer_engine/pytorch/__init__.py
+++ b/transformer_engine/pytorch/__init__.py
@@ -12,6 +12,7 @@ import importlib.util
 import sys
 import torch
 from importlib.metadata import version
+import warnings

 from transformer_engine.common import get_te_path, is_package_installed
 from transformer_engine.common import _get_sys_extension
@@ -78,13 +79,6 @@ from transformer_engine.pytorch.attention import DotProductAttention
 from transformer_engine.pytorch.attention import InferenceParams
 from transformer_engine.pytorch.attention import MultiheadAttention
 from transformer_engine.pytorch.transformer import TransformerLayer
-from transformer_engine.pytorch.permutation import (
-    moe_permute,
-    moe_permute_with_probs,
-    moe_unpermute,
-    moe_sort_chunks_by_index,
-    moe_sort_chunks_by_index_with_probs,
-)
 from transformer_engine.pytorch.fp8 import fp8_autocast
 from transformer_engine.pytorch.fp8 import fp8_model_init
 from transformer_engine.pytorch.graph import make_graphed_callables
@@ -98,3 +92,21 @@ try:
     torch._dynamo.config.error_on_nested_jit_trace = False
 except AttributeError:
     pass  # error_on_nested_jit_trace was added in PyTorch 2.2.0
+
+try:
+    from transformer_engine.pytorch.permutation import (
+        moe_permute,
+        moe_permute_with_probs,
+        moe_unpermute,
+        moe_sort_chunks_by_index,
+        moe_sort_chunks_by_index_with_probs,
+    )
+except RuntimeError as e:
+    if "0 active drivers" in str(e):
+        warnings.warn(
+            "Failed to import permutation functions. "
+            "This is expected if you are importing te on a machine without GPUs.",
+            RuntimeWarning,
+        )
+    else:
+        raise
